app/controllers/devices_controller.rb

```diff
@@ -0,0 +1,41 @@
+class DevicesController < ApplicationController
+  # Devices can only be created for authenticated users.
+  # This replaces the Authenticated Users Pusher Beams solution.
+  # See: https://github.com/forem/forem/pull/12419/files#r563906038
+  before_action :authenticate_user!, only: [:create]
+
+  rescue_from ActiveRecord::ActiveRecordError do |exc|
+    render json: { error: exc.message, status: 422 }, status: :unprocessable_entity
+  end
+
+  def create
+    device = Device.find_or_create_by(device_params)
+    if device.persisted?
+      head :ok
+    else
+      render json: { error: device.errors_as_sentence }, status: :bad_request
+    end
+  end
+
+  def destroy
+    device = Device.find_by(device_params)
+    device&.destroy
+
+    if device&.destroyed?
+      head :no_content
+    else
+      render json: { error: device.errors_as_sentence }, status: :bad_request
```

Comment:
This is going to fail if `find_by` doesn't find anything. `device&.destroyed?` returns `nil`, thus goes in the `else`, thus failing because `nil.errors_as_sentence` doesn't exist.

I think we should make it so that it either returns 404, by using `find_by!` or we return nothing as the error in the `:bad_request`

Reply:
Good point: I reworked this coded to return a 404 if the specified device isn't found, otherwise continue on with the existing statement.